### PR TITLE
Update release instructions for 4.6-maintenance

### PR DESCRIPTION
**Pull request by asottile**
I'm going to run these and make sure they make sense

### CODE REVIEW

File: HOWTORELEASE.rst [hunk 1]
```diff
@@ -33,7 +36,7 @@ taking a lot of time to make a new one.
 
    Wait for the deploy to complete, then make sure it is `available on PyPI <https://pypi.org/project/pytest>`_.
 
-#. Merge the PR into ``master``.
+#. Merge the PR.
```
Comment (nicoddemus) [inline HOWTORELEASE.rst]:
```suggestion
#. Merge the PR.

#. If this is a maintenance release, cherry-pick the commit that updates the CHANGELOG into `master`, while keeping the `changelog` files intact so they get included in the next release as well.
```

Hmmm this is a little annoying I guess... 🤔

Comment (asottile) [reply HOWTORELEASE.rst]: I think I can put this in command form so it's a little easier 🤔

Comment (blueyed) [reply HOWTORELEASE.rst]: Wouldn't/shouldn't changelog updates happen by merging maintenance into master?

Comment (asottile) [reply HOWTORELEASE.rst]: maintenance is intentionally divergent from `master` / `features`.  We actually want all of the changelogs that happen in 4.6 to additionally be part of the 5.x releases

So what we ~basically want to do is cherry pick the changes to `CHANGELOG.rst` and to `doc/en/announce`

**Comment by asottile** [comment]
updated the instructions and used them to create #5365

**Pull request merged**